### PR TITLE
 bugfix/lattice 2284 retry s3 uploads

### PR DESCRIPTION
**Pull request by pckeyes**
This pr adds retry configurations to the s3 client and implements aws's TransferManager for multipart uploads on large (> 16MB) uploads

### CODE REVIEW

File: src/main/kotlin/com/openlattice/data/storage/aws/AwsBlobDataService.kt [hunk 1]
```diff
@@ -43,8 +53,10 @@ class AwsBlobDataService(
         metadata.contentLength = dataInputStream.available().toLong()
         metadata.contentType = contentType
         val putRequest = PutObjectRequest(datastoreConfiguration.bucketName, s3Key, dataInputStream, metadata)
-        s3.putObject(putRequest)
-
+        val transferManager = TransferManagerBuilder.standard().withS3Client(s3).build()
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/data/storage/aws/AwsBlobDataService.kt]: I think you want a `.withMultiPartUploadThreshold(16_000_000L)` here

Comment (pckeyes) [reply src/main/kotlin/com/openlattice/data/storage/aws/AwsBlobDataService.kt]: I believe TransferManager defaults to do multipart upload at 16MB so it doesn't need to be explicitly set

Comment (makosblade) [reply src/main/kotlin/com/openlattice/data/storage/aws/AwsBlobDataService.kt]: Ah yeah they don't put that in the comments anywhere in TransferManager or TMBuilder 🤦‍♂

**Comment by geekbeast** [comment]
This is good to go-- just waiting on benchmarks and will merge.

**Pull request closed without merging**